ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/ConsumptionSpeedMetric.scala

```diff
@@ -11,57 +11,120 @@ import java.time.Instant
 final case class ConsumptionSpeedMetric[T](
     periodMillis: Long,
     recordTimeFunction: T => Seq[Timestamp],
-    firstRecordTime: Option[Instant] = None,
-    lastRecordTime: Option[Instant] = None,
+    objectives: Map[ServiceLevelObjective[ConsumptionSpeedMetric.Value], Option[
+      ConsumptionSpeedMetric.Value
+    ]],
+    previousLatest: Option[Instant] = None,
+    currentPeriodLatest: Option[Instant] = None,
 ) extends Metric[T] {
+  import ConsumptionSpeedMetric._
 
-  override type Value = ConsumptionSpeedMetric.Value
+  override type V = Value
+  override type Objective = ServiceLevelObjective[Value]
 
   override def onNext(value: T): ConsumptionSpeedMetric[T] = {
     val recordTimes = recordTimeFunction(value)
-    val updatedFirstRecordTime =
-      firstRecordTime match {
+    val newPreviousLatest =
+      previousLatest match {
         case None =>
           recordTimes.headOption.map { recordTime =>
```

Comment:
I didn't check where this is coming from, but is `recordTimes` sorted descending or does it grow like a stack? It's a bit confusing to see the head value named as `..latest`

Reply:
From what I understand streams served by a ledger have ascending record times.
I've redefined the `ConsumptionSpeecMetric` a bit. Previously we computed a periodic speed like:
`periodicSpeed = (lastRecordTimeInPeriod - firstRecordTimeInPeriod) / periodDuration`. This definition was problematic in case of a single element in a period or no elements at all.
Currently the definition is: `periodicSpeed = (previousLastConsideredElement - lastConsideredElement) / periodDuration`. The `previousLastConsideredElement` may be in the current period (for the first period) or in the most recent previous period with elements.
I'll update the readme file with detailed description of this.